assets/js/modules/analytics-4/datastore/webdatastreams.js

```diff
@@ -182,7 +182,7 @@ const baseActions = {
 		const registry = yield Data.commonActions.getRegistry();
 		return registry
 			.select( MODULES_ANALYTICS_4 )
-			.getMatchingWebDataStream( propertyID );
+			.getMatchingWebDataStreamByPropertyID( propertyID );
```

Comment:
I noticed that this results in the web data stream to be interpreted as an action (given it's `type` property) which in turn dispatches it since this is seen as a Redux action returned from a generator.

I'm not sure this causes any problems as the type isn't handled so it shouldn't do anything, but it could if we were to later introduce an action by the same type in the future. It's probably okay here since this isn't new but we should probably open an issue to update this so that it returns a non-action-like value.

Reply:
Very interesting find. I've created #6326 to address this. Thank you!